src/components/Accordion/PAccordion.vue

```diff
@@ -0,0 +1,91 @@
+<template>
+  <div class="p-accordion">
+    <template v-for="section in sections" :key="section">
+      <div class="p-accordion__section">
+        <div class="p-accordion__header">
+          <slot :name="`${section}-heading`">
+            <slot name="heading" :section="section">
+              {{ section }}
+            </slot>
+          </slot>
+          <p-button
+            size="sm"
+            icon="ChevronDownIcon"
+            flat
+            rounded
+            class="p-accordion__button"
+            :class="getButtonClass(String(section))"
```

Comment:
I see you casting the `section` to string in quite a few places here - can you explain why that's necessary? I see that these methods expect a string but I'd expect an iterated `Readonly<string[]>` to leave you with a `string`

Reply:
I buy that - thanks for the explanation (and also for pioneering the SFC generics support for this lib)

Reply:
While updating this I realized I could simplify the generics I had and remove the need for the string casting. I had a more complex generic to try and support `defineSlots` but ultimately abandoned the slot types because of the complexity it added and also it just wouldn't compile.